### ISSUE TITLE
Enable Line Segments from Multiple Directions in `create.scatterplot` Function

### ISSUE THREAD

**Issue by maotian06** [uclahs-cds/public-R-BoutrosLab-plotting-general#131]
The current `create.scatterplot` function has the option to add line segments using the following option:
 ```
add.line.segments = FALSE,  line.start = NULL, line.end = NULL
```
However, this function only allows adding the vertical line from the same location of `x`, but not the horizontal lines; here is the source code:
```
    if (add.line.segments) {
      for (i in 1:length(line.start)) {
        with(data = new.env(), expr = panel.segments(x, 
                                                     line.start[[i]][subscripts], x, line.end[[i]][subscripts], 
                                                     col = line.col[[i]], lwd = line.lwd[[i]], lineend = 1))
      }
```
I think the current version doesn't allow for the customization of `x` effectively. It would be useful to enhance the functionality by allowing the addition of multiple arguments, enabling features like drawing horizontal lines and lines from different positions, such as:
```
line.start.x = list(), 
line.start.y = list(), 
line.end.x = list(), 
line.end.y = list()
```

One potential application for this enhancement is in drawing enriched gene locations in the GSEA plot:
![image](https://github.com/uclahs-cds/public-R-BoutrosLab-plotting-general/assets/47042983/771addc4-df9d-494c-9914-50b843a6c886)

**Comment by WuSelina** [comment]
Yes, this! I was also spending a lot of time trying to verify if line segments can only be drawn vertically, because I was trying to do horizontal segments.

**Comment by pboutros** [comment]
This is a great example of a feature that we probably **shouldn't** add.  There is another way to add arbitrary lines to a plot, which is by adding a `panel.abline()` directly to the final trellis object.  For example, you could use the `layer()` function in `latticeExtra`.  We don't want the panel function of `create.scatterplot()` or any equivalent to continually expand, but rather want people to use it for most cases, and then add on top of it using once of the standard `trellis` or `lattice` functionalities.
